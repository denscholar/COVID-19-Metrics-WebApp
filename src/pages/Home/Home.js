import React, { useEffect, useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import { useSelector, useDispatch } from 'react-redux';
import Countries from '../../components/Countries/Countries';
import SearchInput from '../../components/SearchInput/SearchInput';
import HeroPage from '../../components/HeroPage/HeroPage';
import { getCountryCases } from '../../redux/covid/Covid';

const Home = () => {
  const [search, setSearch] = useState('');
  const allCountries = useSelector((state) => state.covid);
  const dispatch = useDispatch();
  const id = uuidv4();

  const fetchCoutries = async () => {
    const res = await fetch('https://disease.sh/v3/covid-19/countries');
    const data = await res.json();
    const countries = data.map((country) => ({ ...country, id }));
    dispatch(getCountryCases(countries));
  };

  const filterCountry = allCountries.filter((country) => country.country.toLowerCase()
    .includes(search.toLowerCase()));

  useEffect(() => {
    if (!allCountries.length) {
      fetchCoutries();
    }
  }, []);

  const handleChange = (e) => {
    e.preventDefault();
    setSearch(e.target.value);
  };

  return (
    <div>
      <HeroPage />
      <SearchInput placeholder="Search country cases" onChange={handleChange} />
      <Countries allCountries={filterCountry} />
    </div>
  );
};

export default Home;
