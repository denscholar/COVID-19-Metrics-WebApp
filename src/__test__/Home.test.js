import React from 'react';
import { create } from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Home from '../pages/Home/Home';

afterEach(() => cleanup());

it('App Renders Correctly', () => {
  const tree = create(
    <Provider store={store}>
      <Home />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
