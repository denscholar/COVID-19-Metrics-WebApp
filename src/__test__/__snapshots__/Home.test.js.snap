// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Correctly 1`] = `
<div>
  <div
    className="sc-dkPtRN dmaGeM container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div>
          <h4>
            Global COVID-19 Metrics
          </h4>
          <ul>
            <li>
              Active:
               
              <span
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                
              </span>
            </li>
            <li>
              Deaths:
               
              <span
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                
              </span>
            </li>
            <li>
              Recovered:
               
              <span
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                
              </span>
            </li>
          </ul>
          <p>
            Last Update:
             
            Invalid Date
          </p>
        </div>
      </div>
    </div>
  </div>
  <nav
    className="input"
  >
    <input
      onChange={[Function]}
      placeholder="Search country cases"
      type="search"
    />
  </nav>
  <div
    className="country-list"
  />
</div>
`;
